twisted/internet/inotify.py

```diff
@@ -137,6 +137,7 @@ def _notify(self, filepath, events):
         """
         Callback function used by L{INotify} to dispatch an event.
         """
+        filepath = filepath.asBytesMode()
```

Comment:
I don't know if there is a better approach, but this was the easiest way for me to get my mind around how to do this, and still have something that worked.
I'm not used to debugging callbacks like this, so it gets a bit convoluted.